### PR TITLE
Add a generic expose function

### PR DESCRIPTION
**Pull request by utkarsh-pro**
**Description**

This PR adds the functionality within linkerd adpater to be able to expose deployments and services. This PR also adds a function specific for exposing the prometheus deployment on a node port. 

The implementation is a simplified adaptation of [`kubectl expose`](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/expose/expose.go)

Tests done:
- [x] Exposing prometheus from the meshery server REST endpoint (/api/mesh/ops) when adapter is outside the container
- [x] Exposing prometheus from the meshery server REST endpoint (/api/mesh/ops) when adapter is inside the container
- [x] Delete exposing service from the meshery server REST endpoint (/api/mesh/ops) when adapter is inside the container
- [x] Delete exposing service from the meshery server REST endpoint (/api/mesh/ops) when adapter is outside the container

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -12,10 +12,9 @@ import (
 
 const (
 	LinkerdOperation = "linkerd"
-	Development      = "development"
```
Comment (Aisuko) [inline internal/config/config.go:15]: We should talk to @kumarabd, let him confirm this change can be true

File: linkerd/expose/traverser.go [hunk 2]
```diff
@@ -19,11 +20,11 @@ type Traverser struct {
...
-	Logger
+	Logger logger.Handler
 }
 
 // Visit function traverses each of the resource mentioned in the Traverser struct
-func (traverser *Traverser) Visit(f func(runtime.Object, error) error, continueOnError bool) error {
+func (traverser *Traverser) Visit(f func(runtime.Object, string, error) error, continueOnError bool) error {
 	var errs []error
 	for _, res := range traverser.Resources {
 		md := strings.Split(res, " ")
```
Comment (Aisuko) [inline linkerd/expose/traverser.go:30]: Should we add more of the code comment with this in detail? Maybe copy the example of the `res`of `traverser.Resources` is good.

File: internal/config/config.go (hunk 1 above)
Comment (utkarsh-pro) [reply internal/config/config.go:15]: @Aisuko Actually I made this change in accordance with the previous discussion where we concluded that the running application shall stay oblivious of its running environment :sweat_smile:

File: linkerd/expose/traverser.go (hunk 2 above)
Comment (utkarsh-pro) [reply linkerd/expose/traverser.go:30]: @Aisuko I agree, will add more comments here, this callback function is a bit vague at the moment. Will make the changes. Thanks :+1:

**Review by leecalcote — APPROVED** [review]

File: linkerd/expose/traverser.go (hunk 2 above)
Comment (leecalcote) [reply linkerd/expose/traverser.go:30]: Are we ready to merge here or do we have outstanding updates?

Comment (utkarsh-pro) [reply linkerd/expose/traverser.go:30]: @leecalcote, I forgot to close this PR. I moved this PR to meshkit (https://github.com/layer5io/meshkit/pull/18). I have made all the changes there which were suggested here. That function is more generic and was tested to expose grafana and prometheus deployed by istio (as addons) as well as linkerd.

On another note, for PRs like https://github.com/layer5io/meshery/pull/2081 (merged) and https://github.com/layer5io/meshery/pull/2070 (open), we faced the issue to identify that which adapter/service mesh was used to deploy the grafana/prometheus or other addons, can we leverage this `Expose` function to annotate the services that we create which will help us determine which adapter or service mesh was responsible for the service's deployment?

**Pull request closed without merging**